pkg/controller/cluster/cluster_controller.go

```diff
@@ -232,6 +235,9 @@ func (c *clusterController) Run(workers int, stopCh <-chan struct{}) error {
 
 	}, c.resyncPeriod, stopCh)
 
+	// sync cluster's KubeConfig cert expiration date per day.
+	go wait.Until(c.syncKubeConfigExpirationDate, time.Hour*24, stopCh)
```

Comment:
If we check the certificate every 24 hrs, what if the user renewed kubeconfig, the condition of the cluster will not be updated immediately.

Reply:
Nice question, I wanted to sync every time I reconcile the cluster, but would it be too frequent?

> the condition of the cluster will not be updated immediately.

If we want to solve this problem perfectly, we need to sync the expiration time every time we reconcile the cluster, WDYT? Any other ideas?

Reply:
I think it is ok since checking kubeconfig expiration is not time-consuming.

Reply:
Updated, PTAL.